cdap-common/src/test/java/co/cask/cdap/common/zookeeper/coordination/ResourceCoordinatorTest.java

```diff
@@ -236,4 +245,22 @@ public InetSocketAddress getSocketAddress() {
       }
     };
   }
+
+  private Collection<PartitionReplica> waitForChange(BlockingQueue<Collection<PartitionReplica>> assignmentQueue,
```

Comment:
From one of the "fail" test run looks like call to the ``poll(5, TimeUnit.SECONDS);`` returns null which likely because that specified waiting time elapses before an element is available. 
Adding this will retry multiple poll calls.